src/Aspire.Hosting.Azure.CosmosDB/api/Aspire.Hosting.Azure.CosmosDB.cs

```diff
@@ -8,25 +8,45 @@
+    public partial class AzureCosmosDBResource : Azure.AzureProvisioningResource, ApplicationModel.IResourceWithConnectionString, ApplicationModel.IResource, ApplicationModel.IManifestExpressionProvider, ApplicationModel.IValueProvider, ApplicationModel.IValueWithReferences, ApplicationModel.IResourceWithEndpoints, Azure.IResourceWithAzureFunctionsConfig
     {
         public AzureCosmosDBResource(string name, System.Action<Azure.AzureResourceInfrastructure> configureInfrastructure) : base(default!, default!) { }
 
         public Azure.BicepSecretOutputReference ConnectionString { get { throw null; } }
 
         public ApplicationModel.ReferenceExpression ConnectionStringExpression { get { throw null; } }
 
+        public Azure.BicepOutputReference ConnectionStringOutput { get { throw null; } }
+
         public bool IsEmulator { get { throw null; } }
+
+        [System.Diagnostics.CodeAnalysis.MemberNotNullWhen(true, "ConnectionStringSecretOutput")]
+        public bool UseAccessKeyAuthentication { get { throw null; } }
+
+        public void ApplyAzureFunctionsConfiguration(System.Collections.Generic.IDictionary<string, object> target, string connectionName) { }
     }
 
     public static partial class AzureCosmosExtensions
     {
         public static ApplicationModel.IResourceBuilder<AzureCosmosDBResource> AddAzureCosmosDB(this IDistributedApplicationBuilder builder, string name) { throw null; }
 
+        public static ApplicationModel.IResourceBuilder<Azure.AzureCosmosDBContainerResource> AddContainer(this ApplicationModel.IResourceBuilder<Azure.AzureCosmosDBDatabaseResource> builder, string name, string partitionKeyPath, string? containerName = null) { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<Azure.AzureCosmosDBDatabaseResource> AddCosmosDatabase(this ApplicationModel.IResourceBuilder<AzureCosmosDBResource> builder, string name, string? databaseName = null) { throw null; }
+
+        [System.Obsolete("This method is obsolete because it has the wrong return type and will be removed in a future version. Use AddCosmosDatabase instead to add a Cosmos DB database.")]
         public static ApplicationModel.IResourceBuilder<AzureCosmosDBResource> AddDatabase(this ApplicationModel.IResourceBuilder<AzureCosmosDBResource> builder, string databaseName) { throw null; }
 
         public static ApplicationModel.IResourceBuilder<AzureCosmosDBResource> RunAsEmulator(this ApplicationModel.IResourceBuilder<AzureCosmosDBResource> builder, System.Action<ApplicationModel.IResourceBuilder<Azure.AzureCosmosDBEmulatorResource>>? configureContainer = null) { throw null; }
 
+        [System.Diagnostics.CodeAnalysis.Experimental("ASPIRECOSMOS001", UrlFormat = "https://aka.ms/dotnet/aspire/diagnostics#{0}")]
```

Comment:
@eerhard this is not in https://learn.microsoft.com/en-us/dotnet/aspire/diagnostics/overview#aspirecosmos001 and I don't see a docs bug.

Reply:
also probably should be ASPIRECOSMOSDB001 ..

Reply:
> and I don't see a docs bug.

[Add docs for CosmosDB vnext-preview emulator (dotnet/docs-aspire#2376)](https://github.com/dotnet/docs-aspire/issues/2376)

Reply:
https://github.com/dotnet/aspire/pull/7689